.github/workflows/ci.yaml

```diff
@@ -160,21 +109,102 @@ jobs:
+    steps:
+      - id: checkout
+        name: Check out source code
+        uses: actions/checkout@v2
+
+      - id: set-up-creds
+        name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-1
+
+      - id: login-ecr
+        name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Download image
+        uses: ishworkh/docker-image-artifact-download@v1
+        with:
+          image: ${{ format('{0}/{1}:{2}-r', steps.login-ecr.outputs.registry, needs.build-docker.outputs.repo-name, needs.build-docker.outputs.tag) }}
+
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+
+      - id: cache-seek
+        name: Check for pip cache hit
+        uses: actions/cache@v2
+        with:
+          # This path is specific to Ubuntu
+          path: ~/.cache/pip
+          # Look to see if there is a cache hit for the corresponding requirements file
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+
+      - id: install
+        name: Install dependencies
+        run: |-
+          python -m pip install --upgrade pip setuptools wheel
+          pip install -r requirements.txt
+      
+      - id: start-container
+        name: Start R container in background
         run: |-
-          echo Pushing image $IMAGE_NAME-builder to ECR.
-          docker push $IMAGE_NAME-builder
+          docker run --rm --name worker-r \
+            -d -p 4000:4000 $R_IMAGE_NAME
+        env:
+          R_IMAGE_NAME: ${{ format('{0}/{1}:{2}-r', steps.login-ecr.outputs.registry, needs.build-docker.outputs.repo-name, needs.build-docker.outputs.tag) }}
 
-          echo Pushing image $IMAGE_NAME to ECR.
-          docker push $IMAGE_NAME
+      - id: test
+        name: Run unit tests
+        run: |-
+          cd src/
+          gunzip -k ../../data/test/python.h5ad.gz
+          CLUSTER_ENV="test" python -m pytest --cov=.
         env:
-          IMAGE_NAME: ${{ format('{0}/{1}:{2}-{3}', steps.login-ecr.outputs.registry, steps.ref.outputs.repo-name, steps.ref.outputs.tag, matrix.project) }}
+          AWS_DEFAULT_REGION: "eu-west-1"
+          CLUSTER_ENV: "test"
+
+      # - id: push
+      #   name: Push docker images to ECR
+      #   run: |-
+      #     echo Pushing image $IMAGE_NAME-builder to ECR.
+      #     docker push $IMAGE_NAME-builder
+
+      #     echo Pushing image $IMAGE_NAME to ECR.
+      #     docker push $IMAGE_NAME
+      #   env:
+      #     IMAGE_NAME: ${{ format('{0}/{1}:{2}-{3}', steps.login-ecr.outputs.registry, steps.ref.outputs.repo-name, steps.ref.outputs.tag, matrix.project) }}
```

Comment:
you will need to remove this commented code, unless we are indeed supposed to push the image to ECR from here? If that is the case, you should also add a condition to push only if it is a pull request or a master commit

Reply:
I modified that! Should be fine now. 

The tests are still not working though so I'll keep looking into this